### DISCUSSION TITLE
Nebula-windows-amd64 is detected as Malware by Microsoft defender

### DISCUSSION THREAD

**Discussion by Syncnetworkteam** [slackhq/nebula#898] [General]
Hi Team, 

We have planned to implement Nebula in our environment. Microsoft Defender has identified and removed the Nebula latest version [Release v1.7.2] as malware. 

Please see the detection snap for more details.

Is it safe to proceed further? since, even in virus total file is defects malware. 

![MicrosoftTeams-image (23)](https://github.com/slackhq/nebula/assets/136309193/0f6fc952-db6d-4752-aa1b-1985c4403bf3)
![MicrosoftTeams-image (24)](https://github.com/slackhq/nebula/assets/136309193/a45cb04d-8ec7-4839-8ae8-f38b40628718)
![MicrosoftTeams-image (25)](https://github.com/slackhq/nebula/assets/136309193/de1d557f-e22b-49e9-bb69-194208c35993)

**Comment by nbrownus** [comment]
Not sure why Windows Defender is angry about this release yet. We have submitted to Microsoft and are waiting to hear back.

The easiest way around this that I have found so far is to build `v1.7.2` manually.

**Reply by Syncnetworkteam** [reply]
Hi nbrownus,

I appreciate your prompt reply. Let me know as soon as you hear back from Microsoft. We cannot disregard the Windows Defender warning in our environment in accordance with our policies. 

Once again thank you for your support.

**Comment by Syncnetworkteam** [comment]
Hi Team, 

Is there any update from Microsoft. 
Still Microsoft Defender is identifing and removed the Nebula latest version [Release v1.7.2] as malware.

**Comment by rawdigits** [comment]
Hi there. I actually stumbled upon a solution that may work for folks, and we are considering distributing it this way until the malware misidentification issue is resolved.

You can use a binary packer like upx to compress executables to be smaller on disk. I just tried it on nebula.exe and it passed windows defender without tripping anything.

<img width="1178" alt="image" src="https://github.com/slackhq/nebula/assets/1096409/dfa5d891-5d72-4862-8195-74b74272656a">

**Comment by rawdigits** [comment]
[nebula.exe.zip](https://github.com/slackhq/nebula/files/11908486/nebula.exe.zip)
You are welcome to try this version for yourself in the meantime.

**Reply by Syncnetworkteam** [reply]
HI Rawdigits, 
I appreciate your help.
Malware has been discovered in the supplied Zip file. Please, could you assist me with this?

![image](https://github.com/slackhq/nebula/assets/136309193/8b6c2021-2a64-4a70-b996-7e8f771ec20d)

![image](https://github.com/slackhq/nebula/assets/136309193/0521a625-ea86-467b-a168-adc068effac2)

**Reply by rawdigits** [reply]
Apologies, but I'm not sure how to help. We are continuing to work on resolving this, but the goal was mainly to fix Windows Defender blocking. I see they are detecting on the fact it uses a packing tool, which is only a temporary measure.

**Reply by rawdigits** [reply]
That said, it should work once downloaded. Defender does not appear to block it.

**Reply by Syncnetworkteam** [reply]
Hi Rawdigits,
Still windows Defender blocks the Nebula.
![image](https://github.com/slackhq/nebula/assets/136309193/97cc633f-538f-4890-a3d0-78a744736fea)

**Reply by Syncnetworkteam** [reply]
Hi Team, 

any updates regarding my request?

**Reply by johnmaguire** [reply]
@Syncnetworkteam No updates, we're continuing to work on it. While this is a priority for us, it's also a system out of our control.

**Reply by Syncnetworkteam** [reply]
Hi Team,

any updates regarding my request?

**Reply by Syncnetworkteam** [reply]
Hi Team, 

Still, we are facing the issue. We need your assistance on this case.

![image](https://github.com/slackhq/nebula/assets/136309193/c333d0a2-3179-47e8-be77-b2bb33a5da4d)


![image](https://github.com/slackhq/nebula/assets/136309193/267426a0-b6ea-4f64-9933-d3df405454bb)

**Reply by Syncnetworkteam** [reply]
![image](https://github.com/slackhq/nebula/assets/136309193/7bb1bc75-bda7-4f6c-b624-79f8a24bb894)

**Reply by johnmaguire** [reply]
We are aware of the issue and continuing to work on a resolution. We will provide an update as soon as we have one.

**Comment by johnmaguire** [comment]
Hi @Syncnetworkteam - this issue should now be resolved. Please see https://github.com/slackhq/nebula/issues/902#issuecomment-1710327589 for complete details. Thanks.
